ADRs/0011-Python-Path-Resolution.md

```diff
@@ -0,0 +1,306 @@
+        version: 3.7.7
+    
+    -   name: py38
+        location: C:\Users\shams\miniconda3\envs\py38
+        version: 3.8.1
+    -------------------------------------------------------------------------------------------------
+
+-   id: 2
+    location: C:\Users\shams\miniconda2\Scripts\conda.exe
+    version: 4.7.1
+    ----------------------------------------ENVIRONMENTS---------------------------------------------
+    -   name: base
+        location: C:\Users\shams\miniconda2
+        version: 2.6.1
+    
+    -   name: py27
+        location: C:\Users\shams\miniconda2\envs\py27
+        version: 2.7.7
+    
+    -   name: py25
+        location: C:\Users\shams\miniconda2\envs\py25
+        version: 2.5.1
+    -------------------------------------------------------------------------------------------------
+-----------------------------------------------------------------------------------------------------
+```
+
+Based on the above info (gained from the command: `konduit pythonpaths`), it's much easier to specify python paths in configuration files (using `type`, `location`, `environment` properties):
+
+```json
+{
+  "@type" : "PYTHON",
+  "pythonConfig" : {
+    
+    "type": "conda",
+    "location": "1",
+    "environment": "py37", 
+
+    "python" : "run.py",
+    "importCodePath" : "imports.py",
+    "pythonInputs" : { },
+    "pythonOutputs" : { },
+    "extraInputs" : { },
+    "returnAllInputs" : false,
+    "setupAndRun" : false,
+    "ioInputs" : {
+      "images" : {
+        "name" : "images",
+        "pythonType" : "list",
+        "secondaryType" : "NDARRAY",
+        "type" : "LIST"
+      }
+    },
+    "ioOutputs" : {
+      "texts" : {
+        "name" : "texts",
+        "pythonType" : "list",
+        "secondaryType" : "STRING",
+        "type" : "LIST"
+      },
+      "boxes" : {
+        "name" : "boxes",
+        "pythonType" : "list",
+        "secondaryType" : "NDARRAY",
+        "type" : "LIST"
+      }
+    },
+    "jobSuffix" : "konduit_job"
+  }
+}
+```
+
+#### Possible Values
+- The `type` property will take one of the following values: [`python`, `conda`, `custom`]
+- `location` will have the *id* of the python install for `type==python` or the id of the conda install for `type==conda`. Otherwise, it will contain the absolute location of the python executable file when `type==custom`
+- `environment` will be looked into if `type==conda` to locate the python installation for a particular conda environment.
+                               
+### Defaults and Priorities
+The defaults for python paths are to be expected in the following way:
+
+![Python Paths Defaults](../markdown_images/python-installation-paths-defaults.png)
+
+### Usage with Profiles
+The profiles should be extended to incorporate the python paths settings for running with the `serve` command. The same items names that are used for the PythonStep config are going to be used here. For example: 
+
+```shell script
```

Comment:
Specify a bit  about how this will work? Also if we're pointing at a conda environment one thing we might want to do is specify a conda environment to use?

Reply:
Yup the shell command will take care of that. You'll have  an option to specify a conda environment name as well when you put `type == conda` in the config.

Reply:
so this command has that option here: https://github.com/KonduitAI/konduit-serving/pull/469/files#diff-87058a0e57d0b969b3864dc5a0e0cbe8R136 under `--conda-env` or `-ce` flag